testing/go/functions_test.go

```diff
@@ -153,3 +153,120 @@ func TestFunctionsMath(t *testing.T) {
 		},
 	})
 }
+
+func TestSystemInformationFunctions(t *testing.T) {
+	RunScripts(t, []ScriptTest{
+		{
+			Name:     "current_database",
+			Database: "test",
+			Assertions: []ScriptTestAssertion{
+				{
+					Query: `SELECT current_database();`,
+					Expected: []sql.Row{
+						{"test"},
+					},
+				},
+				{
+					Query:       `SELECT current_database;`,
+					ExpectedErr: `column "current_database" could not be found in any table in scope`,
+				},
+				// TODO: Implement table function for current_database
+				{
+					Query: `SELECT * FROM current_database();`,
+					Skip:  true,
+					Expected: []sql.Row{
+						{"test"},
+					},
+				},
+				{
+					Query:       `SELECT * FROM current_database;`,
+					ExpectedErr: "table not found: current_database",
+				},
+			},
+		},
+		{
+			Name:     "current_catalog",
+			Database: "test",
+			Assertions: []ScriptTestAssertion{
+				{
+					Query: `SELECT current_catalog;`,
+					Expected: []sql.Row{
+						{"test"},
+					},
+				},
+				{
+					Query:       `SELECT current_catalog();`,
+					ExpectedErr: `ERROR: at or near "(": syntax error (SQLSTATE XX000)`,
+				},
```

Comment:
Interesting, this is a function but it only works without the parenthesis. Not the kind of behavior I would have expected, but perhaps it's mandated by the SQL standard.

Reply:
Yeah I found this note about it here https://www.postgresql.org/docs/9.1/functions-info.html
<img width="681" alt="Screenshot 2024-06-17 at 3 45 41 PM" src="https://github.com/dolthub/doltgresql/assets/29443194/d7086147-7ad0-44ce-84e7-d6ec466fb8a4">